### ISSUE TITLE
Race condition in container creation using devicemapper "no such file or directory"

### ISSUE THREAD

**Issue by AaronDMarasco-VSI** [moby/moby#19296]
Would be a dupe of #4036, but was told to open a new bug report.

Using CentOS7:

```
$ docker info | grep Storage
Storage Driver: devicemapper
$ docker info |grep -i "udev.*sync"
 Udev Sync Supported: true
$ docker info | grep -i loop
(no response)
```

This is an example of the error I receive:

```
docker run -d -v /opt/Xilinx/:/opt/Xilinx/:ro -v /data/jenkins_workspaces/dds:/build --name jenkins-dds-460 jenkins/build:v2-C6
5eadc30da8d1f60a13db3392ba751b197079c579a3dbba43f6a092269a475320
Error response from daemon: Cannot start container 5eadc30da8d1f60a13db3392ba751b197079c579a3dbba43f6a092269a475320: Error getting container 5eadc30da8d1f60a13db3392ba751b197079c579a3dbba43f6a092269a475320 from driver devicemapper: open /dev/mapper/docker-253:6-101737821-5eadc30da8d1f60a13db3392ba751b197079c579a3dbba43f6a092269a475320: no such file or directory
```

A few days ago, I had Docker totally trash all its containers. I researched some, and learned that on CentOS 7 hosts, the preferred method is a "raw" LVM thin provisioned, so that's what I did:

```
$ cat /etc/systemd/system/docker.service.d/0-move-library.conf
[Service]
ExecStart=
ExecStart=/usr/bin/docker daemon --exec-opt native.cgroupdriver=cgroupfs -H fd:// --storage-driver devicemapper --storage-opt dm.fs=xfs --storage-opt dm.thinpooldev=/dev/mapper/vg_ex-docker--pool --storage-opt dm.use_deferred_removal=true

# Needs newer systemd than CentOS uses: https://bugzilla.redhat.com/show_bug.cgi?id=1200946
# --storage-opt dm.use_deferred_deletion=true
```

I'm not listing here all the LVM stuff I did, but it was working for a few days...
## docker info

```
$ cat /etc/redhat-release 
CentOS Linux release 7.1.1503 (Core) 

$ uname -a
Linux redacted.example.com 3.10.0-229.20.1.el7.x86_64 #1 SMP Tue Nov 3 19:10:07 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

$ docker --version
Docker version 1.9.1, build a34a1d5

$ docker info
Containers: 9
Images: 35
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: vg_ex-docker--pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 20.51 GB
 Data Space Total: 322.1 GB
 Data Space Available: 301.6 GB
 Metadata Space Used: 14.26 MB
 Metadata Space Total: 4.001 GB
 Metadata Space Available: 3.987 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 56
Total Memory: 251.6 GiB
```

It definitely seems to be a race condition, when my Jenkins tries to spawn a handful of containers at once. I manually tried this on the server:

```
for i in $(seq 1 25); do
  docker run -d -v /opt/Xilinx/:/opt/Xilinx/:ro -v /data/jenkins_workspaces/examples:/build --name jenkins-examples-485 jenkins/ocpibuild:v2-C6; 
  echo $?; 
  docker rm -f jenkins-examples-485; 
done
```

And it ran flawlessly. However, if I background them instead, to force all 25 to try to create at once:

```
for i in $(seq 1 25); do 
  docker run -d -v /opt/Xilinx/:/opt/Xilinx/:ro -v /data/jenkins_workspaces/examples:/build --name jenkins-ctests-474-${i} jenkins/ocpibuild:v2-C6 & 
  echo $?; 
done
b66c5c2276b40854fc9f7b21e6bf153d861a2493638fb277195ad6a9376f8d32
621dc1b143f1a8728fb93a8227231a7b4488df0d96368e65d21956a6d6272004
6c6d1b417cd343469804c16f4d8479775f3aba905014d72cdb061f141f2ba816
127fdfa18b8d332da7a26e9cd9cb2f49ee4d4a000cf7ea2fe20858fffe1d7a09
66c0100da826fb6a94c084b3f4d9c53583b3e699e8a26076035c0f09bf61ec30
d16237d791da495d711483dc02c7021ab42781c001eb46128a064b3f1ddb245e
72176e1075e4a837ae7abeee9bfaee30b2f08f27cfa3542a66ebcd6a51115326
7d10de31239df16bca6c69d31d151d5ad7f5e44036f98e394e14fa9853408824
a582799ff489507fd13b02b3e2f2032a93c7f1bfbc88d443fee4526b8bddf84d
84de8f525c76ca1b64ea0c25f63a885a0e35ffa0cf2493361c56270635149023
0f104d795cb386542f0850bf964a5f426fe01476d8b56a414fdd98c6f145a973
1ad7c18178d54a53fa4fa4cda67de75923af1c965527a713542f73ac8a58cc52
ce153bb43b1fdb4ad193e72d36e8451baedf97c8c81bbe18f19818f7492eaf74
29ccbf74238de02c96eeee1eabd2f45e9163ab0633d96465b76d10a83e81f947
f178c90fb051303180bfcd0cde370526bd4a96f37ed94b3bcf56350d989e267f
46e91895d8c5c475bee7db2dc1a4c0592d4a715628dce807eff353739c7aeb4e
4c36736d89d06d62509a70203c66b74dd6ce37e7de6e6c5a2ffb3174525eed0d
86a913665de1522922726e901a54800414cc9858679168228fa626a255704d4c
c38da55732a1ff5df62238391b042a2129ad701ab2ddb826f99c81326bf26f74
57b6e0356ca25b3746c680bd6a535c514d64f4bcb50b900b9a93018207034b72
7651b72be21fb358b3fc01bd006b6e0c8fa8eb958e1cb1719e9ac223e214770e
521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a
Error response from daemon: Cannot start container 7651b72be21fb358b3fc01bd006b6e0c8fa8eb958e1cb1719e9ac223e214770e: Error getting container 7651b72be21fb358b3fc01bd006b6e0c8fa8eb958e1cb1719e9ac223e214770e from driver devicemapper: open /dev/mapper/docker-253:6-101737821-7651b72be21fb358b3fc01bd006b6e0c8fa8eb958e1cb1719e9ac223e214770e: no such file or directory
Error response from daemon: Cannot start container 521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a: Error getting container 521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a from driver devicemapper: open /dev/mapper/docker-253:6-101737821-521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a: no such file or directory
ac57455b5030d56f163b45e7d1f05856c0429823fc708b37aa21b5925f70da37
```

So 2 of 25 failed.

And they "stay dead" - I can never start them, even though their status is "Created":

```
$ docker ps -a | grep 521cc795b9
521cc795b9e9        jenkins/ocpibuild:v2-C6   "/bin/sleep 1h"     2 minutes ago       Created                                       jenkins-ctests-474-3
$ docker start jenkins-ctests-474-3
Error response from daemon: Cannot start container jenkins-ctests-474-3: Error getting container 521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a from driver devicemapper: open /dev/mapper/docker-253:6-101737821-521cc795b9e9f65cc5651a457593b540369188e17281657488490e36b62f370a: no such file or directory
Error: failed to start containers: [jenkins-ctests-474-3]
```

**Comment by AaronDMarasco-VSI** [comment]
Here's my workaround for now:

```bash
set +e
for i in $(seq 1 10); do
  if [ 10 == $i ]; then set -e; fi;
  docker run -d -v /opt/Xilinx/:/opt/Xilinx/:ro -v ${WORKSPACE}:/build --name ${BUILD_TAG} jenkins/ocpibuild:v2-C${TARGET_OS}
  if [ $? -eq 0 ]; then
    break
  else
    echo "Docker failed to launch ($i)"
    docker rm -f ${BUILD_TAG} || :
  fi
done
```

**Comment by rhvgoyal** [comment]
I ran following on fedora 23 twice (with docker 1.10-dev) and I could not reproduce the problem. Is there any chance you can try running on fedora 23 and see if you can still reproduce the problem.

for i in $(seq 1 25); do
  docker run -d --name race-${i} fedora bash &
  echo $?;
done

**Comment by AaronDMarasco-VSI** [comment]
No, I cannot switch OSs on this server and don't have a spare with the same configuration. @RoelVdP noted he hit it as well; not sure what his config was. Sorry!

**Comment by rhvgoyal** [comment]
Ok, I created similar configuration as yours and still  I can't reproduce the problem.

**Comment by rhvgoyal** [comment]
`# cat /etc/redhat-release 
CentOS Linux release 7.1.1503 (Core)
`

**Comment by rhvgoyal** [comment]
# docker info
Containers: 25
Images: 2
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker--vg-docker--pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 674.8 MB
 Data Space Total: 5.142 GB
 Data Space Available: 4.467 GB
 Metadata Space Used: 786.4 kB
 Metadata Space Total: 16.78 MB
 Metadata Space Available: 15.99 MB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 2
Total Memory: 3.614 GiB
Name: centos7-generic-2.os1.phx2.redhat.com
ID: UA56:4AIQ:K7UC:2QL5:NXF5:IXEO:RBSL:W2MQ:BWNJ:762T:HLD3:MIOG
Debug mode (server): true
 File Descriptors: 17
 Goroutines: 27
 System Time: 2016-01-15T14:42:52.496560361Z
 EventsListeners: 0
 Init SHA1: cd300e9e14aa378c3e74bada822d3e00a5f7c32b
 Init Path: /root/git/rhvgoyal-docker/bundles/latest/dynbinary/dockerinit
 Docker Root Dir: /var/lib/docker

**Comment by AaronDMarasco-VSI** [comment]
@rhvgoyal Can you get a machine with more CPUs to get more concurrence? 

Sidebar - how did you get "Deferred Deletion Enabled" to work? I was under the impression it needed a [later systemd than RHEL ships](https://bugzilla.redhat.com/show_bug.cgi?id=1200946)?

Those are the only two differences I can quickly spot.

**Comment by rhvgoyal** [comment]
deferred deletion automatically gets enabled by "docker-storage-setup" if docker version is equal to higher than 1.9.

**Comment by AaronDMarasco-VSI** [comment]
OK, I didn't run that because I couldn't find an RPM for it. I built and configured the storage manually.

For the concurrence thing, maybe you can nest it in some way so multiple copies all run at the same time?

Does the devicemapper internally make any assumptions, like "you'll never have more than 32 CPUs?"

**Comment by rhvgoyal** [comment]
I don't think devicemapper makes any assumption about number of cpus being less than 32.

**Comment by rhvgoyal** [comment]
@AaronDMarasco-VSI Can you boot your machine with say nr_cpus=2 to reduce number of actively booted cpus  and see if you are still able to reproduce problem.

**Comment by AaronDMarasco-VSI** [comment]
Yeah, you're going to need to test on a bigger box.

This is what I used to switch CPU count. I confirmed after each reboot that `/proc/cpuinfo` had the "correct" count.

```
sudo grubby --args="nr_cpus=2" --update-kernel /boot/vmlinuz-3.10.0-229.20.1.el7.x86_64
sudo grubby --info /boot/vmlinuz-3.10.0-229.20.1.el7.x86_64
```

Each time I did an iteration of 100 this time:
```
for i in $(seq 1 100); do    docker run -d -v /opt/Xilinx/:/opt/Xilinx/:ro -v /data/jenkins_workspaces/examples:/build --name jenkins-ctests-474-${i} jenkins/ocpibuild:v2-C6 &    echo $?;  done
for a in $(seq 1 100); do docker rm -f jenkins-ctests-474-${a} & done
```

Results (number of failures, presented in order of execution):
2 CPUs: 0, 0
32 CPUs:  2, 0, 1
2 CPUs: 0, 0, 0
16 CPUs... 2 (1 new error "`Error response from daemon: Error running deviceResume dm_task_run failed`"), 0, 0, 0
8 CPUs... 0, 1, 0, 0
48 CPUs... 0, 1, 0, ??? (Ran out of IPv4 addresses, restarted service), 1
56 CPUs (forced the natural count E5-2695v3 = 14 physical cores x 2 physical sockets x 2 HyperThread, forced with kernel command line)... 0, 0, 0, 0 (really?), 0, 1

I am guessing that I'm hitting the bug a lot less because I am constantly rebooting the server so nobody else (like Jenkins) is really stressing it out at the same time...

**Comment by rhvgoyal** [comment]
Ok, so it does not happen all the time and there is higher probability of it happening on bigger box. I will see if I can get hold of a higher cpu count box and retry.

**Comment by AaronDMarasco-VSI** [comment]
Looks like it. I never got it to hit on 2 CPUs. I didn't try with and without external volumes; that's another difference I see.

**Comment by RoelVdP** [comment]
I am quite confident my config would have been 1) Centos 7 x64, 2) 8-thread, single CPU machine. 

[roel@localhost ~]$ cat /proc/cpuinfo | egrep "processor|model name" | tail -n2
processor       : 7
model name      : Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz

From what I saw, the issue was definitely sporadic. Try and overload the machine. 

Also, from my testing, issue reproducibility increased using "build" (much more reproducible with "build" then with "run"). There may be a higher producibility using "rmi" also.

Please also see https://github.com/docker/docker/issues/4036#issuecomment-70048440 and note that I reproduce the bug on the standard centos5 image too previously.

@all: when posting, please use "CPU-threads" (or simply "threads") vs "CPU" - as it's quite a difference.

**Comment by rhvgoyal** [comment]
I just tried centos (1511) with stock docker (1.8.2) on a 143 logical cpu machine and I can't seem to be able to hit the issue in 4-5 attempts so far.

**Comment by AaronDMarasco-VSI** [comment]
@RoelVdP Sorry, I slacked, but when I was saying CPUs, I was explicitly referring to the value given to the kernel with the `nr_cpus` command line argument, since that is what I was asked to test with.

**Comment by rhvgoyal** [comment]
Kernel Version: 3.10.0-327.4.4.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 144
Total Memory: 125.6 GiB

**Comment by RoelVdP** [comment]
@AaronDMarasco-VSI Makes sense :)

**Comment by AaronDMarasco-VSI** [comment]
I am using docker 1.9.1 (a34a1d5) from this repo:
```
$ cat /etc/yum.repos.d/docker-main.repo 
[docker-main-repo]
name=Docker Main Repository
baseurl=https://yum.dockerproject.org/repo/main/centos/7
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg
```
My kernel is also patch 229 (1503).

**Comment by rhvgoyal** [comment]
@AaronDMarasco-VSI Any chance you can move to latest centos relese (1511) and see if the issue still happens. I will move to docker 1.9.1 and retry.

**Comment by rhvgoyal** [comment]
I install dokcer-engine rpm from repo you mentioned above and I am getting bunch of errors. But they are related to cgroups.

Error response from daemon: Cannot start container bffe9851878c914853e9454e5b26130f0bd0cf2d622f0715bfd3006e0060c75a: [8] System error: write /sys/fs/cgroup/devices/system.slice/docker-bffe9851878c914853e9454e5b26130f0bd0cf2d622f0715bfd3006e0060c75a.scope/cgroup.procs: no such device
Error response from daemon: Cannot start container dbb3f5fc07f93c6ba513dda165b6cc493dc568a76ce1686cd972c77bead24a19: [8] System error: open /sys/fs/cgroup/devices/system.slice/docker-dbb3f5fc07f93c6ba513dda165b6cc493dc568a76ce1686cd972c77bead24a19.scope/cgroup.procs: no such file or directory
Error response from daemon: Cannot start container d1d457ace99d3b5951fc0f1eae3cfb19dd01b044593b0e2ebf74341b7f005441: [8] System error: write /sys/fs/cgroup/devices/system.slice/docker-d1d457ace99d3b5951fc0f1eae3cfb19dd01b044593b0e2ebf74341b7f005441.scope/cgroup.procs: no such device

**Comment by AaronDMarasco-VSI** [comment]
@rhvgoyal Add `native.cgroupdriver=cgroupfs` to your `ExecStart` line.

**Comment by rhvgoyal** [comment]
Ok, `--exec-opt native.cgroupdriver=cgroupfs` seems to have solved the cgroup issue. But creating 100 containers in parallel succeeded. Will try 200 now.

CC @mrunalp mrunal, I am curious that why I had to add native.cgroupdriver=cgroupfs to solve cgroup issue.

**Comment by rhvgoyal** [comment]
Starting 200 containers worked fine. Trying 300 now.

**Comment by rhvgoyal** [comment]
Trying to go above 200 clients does not work as docker can't handle so many clients and some clients can't connect. And 200 containers seems to be working well for me.

**Comment by AaronDMarasco-VSI** [comment]
We have the server locked at 7.1 for about another month because we are in the critical phase of our project.

**Comment by mrunalp** [comment]
@rhvgoyal Could you paste the output of /proc/1/cgroup in a container with and without cgroupdriver option?

**Comment by rhvgoyal** [comment]
@mrunalp 

Sure. Following is output without cgroup driver option.

cat /proc/1/cgroup 
10:perf_event:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
9:blkio:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
8:freezer:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
7:devices:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
6:net_cls:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
5:cpuacct,cpu:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
4:hugetlb:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
3:memory:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
2:cpuset:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope
1:name=systemd:/system.slice/docker-56ee699808d646641b40b142c45df6264fc4ef951144ff0d11c0a9ccc40e2c77.scope

**Comment by rhvgoyal** [comment]
Following is output with cgroupdriver option.

10:perf_event:/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
9:blkio:/system.slice/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
8:freezer:/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
7:devices:/system.slice/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
6:net_cls:/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
5:cpuacct,cpu:/system.slice/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
4:hugetlb:/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
3:memory:/system.slice/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
2:cpuset:/docker/09a175c7cd85444425a57da61d649623df46ab03d746b353b2eb705f4bb4db19
1:name=systemd:/system.slice/docker.service

**Comment by AaronDMarasco-VSI** [comment]
I've now upgraded, as requested. Now running CentOS 7.2 (1511, kernel 3.10.0-327.10.1.el7.x86_64) with Docker version 1.10.2, build c3959b1.

I ran into the problem again, but it seemed to be happening a lot less (possibly because I just rebooted):
```
docker: Error response from daemon: open /dev/mapper/docker-253:7-101737821-c27f5fb6f80fbb3ae17e52f67d9c2b9604062c82a9dd1ef5bb34538f8e0fd827: no such file or directory.
docker: Error response from daemon: open /dev/mapper/docker-253:7-101737821-c91b898cea7ca74664842f585bcfde5782119fa6be581f5be433b691d5c9a60f: no such file or directory.
docker: Error response from daemon: open /dev/mapper/docker-253:7-101737821-3775149f534febd85479ac3f4d67406a9b6b2be4acaf6c2f98fa4fbbe97afc05: no such file or directory.
docker: Error response from daemon: open /dev/mapper/docker-253:7-101737821-bde9a4dc039d9ceba7ff051a6b9cc3431efd8974035bd5f3b0f4f27a56139116: no such file or directory.
```
This failed only four times in about 1000 container creations.

So, it looks like things got a _lot_ better, but I can still replicate it on rare occasion.

**Comment by grisha87** [comment]
@AaronDMarasco-VSI, and you have Udev Sync Supported marked as true? ($ docker info)?

**Comment by AaronDMarasco-VSI** [comment]
@grisha87 Yes, that is noted in the first post. I just confirmed that it is still set with v1.10.2.

**Comment by freme** [comment]
I am running on amazon linux 2016.03 and have the same issue. The problem also occurred on the 2015 releases. I use this in jenkins and the propability rises with the runtime of the jenkins slave and the number of parallel jobs.

Docker deamon is started with:
`OPTIONS="-H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --storage-opt dm.basesize=100G --storage-driver=devicemapper --storage-opt dm.metadatadev=/dev/direct-lvm/metadata --storage-opt dm.datadev=/dev/direct-lvm/data --storage-opt dm.blkdiscard=false --storage-opt dm.fs=xfs --storage-opt dm.use_deferred_removal=true --storage-opt dm.use_deferred_deletion=true"`

LVM thinpool is created like this:
`pvcreate /dev/xvdb
vgcreate direct-lvm /dev/xvdb
lvcreate --wipesignatures y -n data direct-lvm -l 97%VG
lvcreate --wipesignatures y -n metadata direct-lvm -l 3%VG`

Only happens every 100 jobs or so, but this means at least once every hour in my case . . .

**Comment by manoj0077** [comment]
Hi, we are also hitting the same issue but not very frequently. Is there a way we can start the container even though it failed without daemon restart ?

**Comment by AaronDMarasco-VSI** [comment]
>  Is there a way we can start the container with some manual steps even though it failed without daemon restart ?

I am not aware of one; I've had to nuke it and start from scratch each time. The first comment is the snippet I've used.

**Comment by brettcs** [comment]
I'm seeing very similar issues on a Debian 8 server, running dynamically linked Docker from Docker's open source repository.  I can semi-reliably reproduce the error by starting many containers near-simultaneously, as suggested in the original report.  My own core test command is just `docker run --name=… debian:8 /bin/sleep 5s &`—I tried to whittle that down as much as possible to avoid any confounding factors.

<pre>$ sudo docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:39 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:39 2016
 OS/Arch:      linux/amd64
$ sudo docker info
Containers: 16
 Running: 0
 Paused: 0
 Stopped: 16
Images: 213
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: docker-thinpool-tpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 30.81 GB
 Data Space Total: 102 GB
 Data Space Available: 71.19 GB
 Metadata Space Used: 9.871 MB
 Metadata Space Total: 1.07 GB
 Metadata Space Available: 1.06 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.90 (2014-09-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 29.52 GiB
ID: 776W:FPWS:B6JO:HYN3:67ON:P67X:SCVR:DVGP:662D:F3PU:Z3E2:3STP
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
</pre>

I recently upgraded this daemon from aufs to devicemapper, following [the instructions](https://docs.docker.com/v1.11/engine/userguide/storagedriver/device-mapper-driver/) as closely as I could.

In case it ends up being relevant, this is also on a Jenkins server, but I can reproduce the problem outside Jenkins.

**Comment by bcdonadio** [comment]
I also could reproduce it on Fedora 24 with direct-lvm. Just once so far, though. If I end hitting this bug further, I will report here.

```
[bcdonadio@darwin zimp-deploy] $ docker version
Client:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-1.10.3-21.git19b5791.fc24.x86_64
 Go version:      go1.6.2
 Git commit:      19b5791/1.10.3
 Built:           
 OS/Arch:         linux/amd64

Server:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-1.10.3-21.git19b5791.fc24.x86_64
 Go version:      go1.6.2
 Git commit:      19b5791/1.10.3
 Built:           
 OS/Arch:         linux/amd64
[bcdonadio@darwin zimp-deploy] $ docker info
Containers: 20
 Running: 10
 Paused: 0
 Stopped: 10
Images: 190
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 10.11 GB
 Data Space Total: 99.61 GB
 Data Space Available: 89.5 GB
 Metadata Space Used: 5.136 MB
 Metadata Space Total: 1.044 GB
 Metadata Space Available: 1.039 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.122 (2016-04-09)
Execution Driver: native-0.2
Logging Driver: journald
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 4.5.7-300.fc24.x86_64
Operating System: Fedora 24 (Workstation Edition)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 0
CPUs: 4
Total Memory: 7.796 GiB
Name: darwin.bcdonadio.com
ID: XRWH:JZWH:C6G5:6AL3:CRZI:XPDX:34LR:6VIK:WW7M:MMFR:4KC5:H2G7
Username: bcdonadio
Registry: https://index.docker.io/v1/
Registries: docker.io (secure)
```

**Comment by wemu** [comment]
we encounter the same issue as well:

```
ERROR: for container_db  Cannot start service container_db: open /dev/mapper/docker-253:3-58720449-67bb2de26671cebc67a8d7d76f00dcedfdc18e1a483ada0396c42862dbf87a4e: no such file or directory
```

Happens on the CI server build agents, about once a day on the 6 build agents. What sometimes helps is to remove the containers, images and volumes. But not every time. Sometimes nothing has to be done - other builds will run normally, just one will not. So it seems to be related to a hanging image.

Docker Info:
```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 12
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: vgapp01-docker--pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 3.025 GB
 Data Space Total: 45.32 GB
 Data Space Available: 42.29 GB
 Metadata Space Used: 1.548 MB
 Metadata Space Total: 117.4 MB
 Metadata Space Available: 115.9 MB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: null host bridge
 Authorization: rhel-push-plugin
Kernel Version: 3.10.0-327.36.2.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.2 (Maipo)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 4
Total Memory: 7.632 GiB
Name: ld9c0274.umobi.mobicorp.test
ID: LVPV:GVKX:PNLT:CPZO:3DPT:5NBZ:643E:XRPE:SEQ4:JMWB:S2NE:KXPX
Registries: docker-registry.company.ch (secure), docker-registry-pz.company.ch (secure), docker.io (secure)
```

**Comment by beatrubi** [comment]
Guys, I was able to create a simple reproducer:

`# for j in cont{00..99}; do docker run --rm --privileged --name=$j --hostname=$j registry.access.redhat.com/rhel7 sh -c "uname -a; sleep 300" & done`

I see roughly 5-10 failing container starts on all docker versions on a VM with 2 vCPUs and 2G memory. I tested so far (1.11, 1.12) on RHEL7, Fedora 25 and Ubuntu 11.04 LTS when using device mapper driver as storage backend.
HTH, Beat

**Comment by AaronDMarasco-VSI** [comment]
@beatrubi That looks a simpler version of my loop that was reported a year (!) ago. Maybe the extra `--rm` helps beat on it. I'll take _anything_ to make this ticket more testable and visible.

**Comment by thaJeztah** [comment]
The devicemapper storage-driver was deprecated, and has been removed, so closing issues related to this storage-driver. See;

- https://github.com/moby/moby/pull/38017
- https://github.com/moby/moby/pull/43637
